### PR TITLE
[sdk#809] Refresh should not happen after failed request

### PR DESCRIPTION
**Pull request by Bolodya1997**
## Description
Disable refresh if request fails.

## Issue link
#809

## How Has This Been Tested?

- [x] Added unit testing to cover
- [ ] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [x] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

**Review by denis-tingaikin — APPROVED** [review]

File: pkg/networkservice/common/refresh/client_test.go [hunk 1]
```diff
@@ -267,3 +268,24 @@ func TestRefreshClient_Sandbox(t *testing.T) {
...
+		refresh.NewClient(ctx),
+		injecterror.NewClient(),
+	)
+
+	_, err := client.Request(ctx, &networkservice.NetworkServiceRequest{
+		Connection: new(networkservice.Connection),
+	})
+	require.Error(t, err)
+
+	goleak.VerifyNone(t)
```
Comment (denis-tingaikin) [inline pkg/networkservice/common/refresh/client_test.go:290]: Why do we need this?
```suggestion
```

Comment (Bolodya1997) [reply pkg/networkservice/common/refresh/client_test.go:290]: To validate that there is no pending refresh action.

Comment (denis-tingaikin) [reply pkg/networkservice/common/refresh/client_test.go:290]: I don't quite follow we already have check on cleanup at line 273

Comment (denis-tingaikin) [reply pkg/networkservice/common/refresh/client_test.go:290]: @Bolodya1997 Could you please explain this line?

File: pkg/networkservice/common/refresh/client_test.go [hunk 2]
```diff
@@ -267,3 +268,24 @@ func TestRefreshClient_Sandbox(t *testing.T) {
 		return refreshSrv.getState() == testRefreshStateDoneRequest
 	}, sandboxTotalTimeout, sandboxStepDuration)
 }
+
+func TestRefreshClient_NoRefreshOnFailure(t *testing.T) {
```
Comment (denis-tingaikin) [inline pkg/networkservice/common/refresh/client_test.go:272]: It seems to me this test is incorrect due to there are no expire time in request.
1. Please add expiration time and use `clockmock` to check that there are no new requests after getting an error on the request. 
2. Please add test for Close use-case. As I can see you've added a major return statement at pkg/networkservice/common/refresh/client.go:133. So this scenario should be covered.

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

File: pkg/networkservice/common/refresh/client_test.go (hunk 1 above)
Comment (denis-tingaikin) [reply pkg/networkservice/common/refresh/client_test.go:290]: Can we use clockmock instead of an additional goleak call?

Comment (Bolodya1997) [reply pkg/networkservice/common/refresh/client_test.go]: Yep, fixed.

File: pkg/networkservice/common/refresh/client_test.go (hunk 2 above)
Comment (Bolodya1997) [reply pkg/networkservice/common/refresh/client_test.go:273]: Yes, added.

**Review by denis-tingaikin — APPROVED** [review]

**Review by d-uzlov — APPROVED** [review]

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**